_inc/lib/admin-pages/class.jetpack-my-jetpack-page.php

```diff
@@ -46,11 +46,34 @@ function jetpack_my_jetpack_change_user() {
 				Jetpack::init()->stat( 'admin', 'change-primary-successful' );
 				Jetpack::init()->do_stats( 'server_side' );
 
+				// Change the blog owner dotcom side
+				$this->tell_dotcom_about_user_switch( $new_master_user );
 			}
 		}
 	}
 
 	/*
+	 * Tell wpcom that the master user has switched
+	 * so we can update the 'wpcom_blog_owner'
+	 */
+	function tell_dotcom_about_user_switch( $new_master ) {
+		$id = Jetpack::get_option( 'id', false );
+		$wpcom_user = Jetpack::get_connected_user_data( $new_master );
```

Comment:
You should be able to pass in just the remote user ID, and the wpcom side can look in the `jetpack_tokens_mu` table by blog id to find out their wpcom id if they are connected to wpcom.